li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/PrimitiveEncoderDecoder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils;
+
+/**
+ * Class to encode/decode primitive integer/long types.
+ */
+public class PrimitiveEncoderDecoder {
+  // The number of bytes for a long variable
+  public static final int LONG_SIZE = Long.SIZE / Byte.SIZE;
+  public static final int INT_SIZE = Integer.SIZE / Byte.SIZE;
+
+  /**
+   * Avoid instantiating PrimitiveEncoderDecoder class
+   */
+  private PrimitiveEncoderDecoder() {
+
+  }
+
+  /**
+   * Encodes a long value into a {@link PrimitiveEncoderDecoder#LONG_SIZE} byte array
+   */
+  public static void encodeLong(long value, byte[] bytes, int pos) {
+    if (bytes == null) {
+      throw new IllegalArgumentException("The input result cannot be null");
+    }
+
+    if (pos < 0) {
+      throw new IllegalArgumentException("position cannot be less than zero");
+    }
+
+    if (bytes.length < pos + LONG_SIZE) {
+      throw new IllegalArgumentException(
+          String.format("Not adequate bytes available to encode the long value(array length = %d, pos = %d", bytes.length, pos)
+      );
+    }
+
+    for (int i = LONG_SIZE - 1; i >= 0; i--) {
+      bytes[pos + i] = (byte) (value & 0xffL);
+      value >>= 8;
+    }
+  }
+
+  /**
+   * Encodes a long value into a newly created byte[] and returns it.
+   */
+  public static byte[] encodeLong(long value) {
+    byte[] data = new byte[LONG_SIZE];
+    encodeLong(value, data, 0);
+    return data;
+  }
+
+  /**
+   * Encodes a int value into a {@link PrimitiveEncoderDecoder#INT_SIZE} byte array
+   */
+  public static void encodeInt(int value, byte[] bytes, int pos) {
+    if (bytes == null) {
+      throw new IllegalArgumentException("The input result cannot be null");
+    }
+
+    if (pos < 0) {
+      throw new IllegalArgumentException("position cannot be less than zero");
+    }
+
+    if (bytes.length < pos + INT_SIZE) {
+      throw new IllegalArgumentException(
+          String.format("Not adequate bytes available to encode the int value(array length = %d, pos = %d", bytes.length, pos)
+      );
+    }
+
+    bytes[pos] = (byte) (value >> 24);
+    bytes[pos + 1] = (byte) (value >> 16);
+    bytes[pos + 2] = (byte) (value >> 8);
+    bytes[pos + 3] = (byte) value;
+  }
+
+  /**
+   * Encodes a int value int a newly created byte[] and returns it
+   */
+  public static byte[] encodeInt(int value) {
+    byte[] data = new byte[INT_SIZE];
+    encodeInt(value, data, 0);
+    return data;
+  }
+
+  public static int decodeInt(byte[] bytes, int pos) {
+    if (bytes == null) {
+      throw new IllegalArgumentException("bytes cannot be null");
+    }
+
```

Comment:
Add pos check to be consistent with others

Reply:
Good catch.